mypyc/irbuild/specialize.py

```diff
@@ -356,3 +358,53 @@ def translate_dict_setdefault(
                               [callee_dict, key_val, data_type],
                               expr.line)
     return None
+
+
+@specialize_function('format', str_rprimitive)
+def translate_str_format(
+        builder: IRBuilder, expr: CallExpr, callee: RefExpr) -> Optional[Value]:
+    if (isinstance(callee, MemberExpr) and isinstance(callee.expr, StrExpr)
+            and expr.arg_kinds.count(ARG_POS) == len(expr.arg_kinds)):
+
+        format_str = callee.expr.value
+        if not can_optimize_format(format_str):
+            return None
+
+        format_str = format_str.replace("{{", "{")
+        format_str = format_str.replace("}}", "}")
```

Comment:
This isn't right? `{{}}` gets translated to `{}`, and we'll split it below, even though we shouldn't do it?

Reply:
Since we would support many operations other than empty brackets, I'll add a helper function for parsing instead of using `replace` and `split`.